src/ringct/rctSigs.cpp

```diff
@@ -335,16 +345,41 @@ namespace rct {
       hashes.push_back(hash2rct(h));
 
       keyV kv;
-      kv.reserve((64*3+1) * rv.p.rangeSigs.size());
-      for (auto r: rv.p.rangeSigs)
+      if (rv.type == RCTTypeSimpleBulletproof)
       {
-        for (size_t n = 0; n < 64; ++n)
-          kv.push_back(r.asig.s0[n]);
-        for (size_t n = 0; n < 64; ++n)
-          kv.push_back(r.asig.s1[n]);
-        kv.push_back(r.asig.ee);
-        for (size_t n = 0; n < 64; ++n)
-          kv.push_back(r.Ci[n]);
+        kv.reserve((6*2+10) * rv.p.bulletproofs.size());
+        for (const auto &p: rv.p.bulletproofs)
+        {
+          CHECK_AND_ASSERT_THROW_MES(p.V.size() == 1, "V has not exactly one element");
```

Comment:
Perhaps do `CHECK_AND_ASSERT_THROW_MES` also for L and R (not empty and the same size)?

Reply:
Or maybe, since this validity checking is also done elsewhere, perhaps the class `rct::Bulletproof` could have a member function that checks this?

Reply:
It doesn't really matter to this function. In fact, I probably should just push all Vs, and let the other code decide on the semantics, that'll be more future proof.